File: frontend/lib/schemas/gemi.ts
import { z } from "zod";

/** Normalized GEMI search item from backend `GemiService._normalize_results` */
export const gemiSearchItemSchema = z
  .object({
    company_name: z.string(),
    afm: z.string(),
    ar_gemi: z.string(),
    legal_type: z.string().optional(),
    zip_code: z.string().optional(),
    municipality: z.string().optional(),
    city: z.string().optional(),
    street: z.string().optional(),
    street_number: z.string().optional(),
    phone: z.string().optional(),
    email: z.string().optional(),
    objective: z.string().optional(),
    status: z.string().optional(),
    gemi_office: z.string().optional(),
  })
  .passthrough();

export const gemiSearchResponseSchema = z.object({
  items: z.array(gemiSearchItemSchema),
  query: z.string(),
  mode: z.enum(["afm", "gemi_number"]),
  exact: z.boolean(),
});

export const gemiFetchDocumentsResponseSchema = z.object({
  company: z.string(),
  documents_uploaded: z.number(),
  message: z.string(),
});

export type GemiSearchItem = z.infer<typeof gemiSearchItemSchema>;
export type GemiSearchResponse = z.infer<typeof gemiSearchResponseSchema>;
export type GemiFetchDocumentsResponse = z.infer<
  typeof gemiFetchDocumentsResponseSchema
>;

File: backend/app/models/gemi.py
"""Pydantic response schemas for GEMI / companies API routes."""
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


class GemiCompanyHit(BaseModel):
    """Single company row from GEMI search, normalized for onboarding UIs."""

    company_name: str = ""
    afm: str = ""
    ar_gemi: str = ""
    legal_type: str = ""
    zip_code: str = ""
    municipality: str = ""
    city: str = ""
    street: str = ""
    street_number: str = ""
    phone: str = ""
    email: str = ""
    objective: str = ""
    status: str = ""
    gemi_office: str = ""


class GemiSearchResponse(BaseModel):
    """Envelope for ``GET /v1/companies/gemi/search``."""

    items: list[GemiCompanyHit] = Field(default_factory=list)
    query: str
    mode: Literal["afm", "gemi_number"]
    exact: bool = False
    warning: str | None = None


class GemiDocumentsFetchResponse(BaseModel):
    """Result of ``POST /v1/companies/gemi/{afm}/documents:fetch``."""

    company: str
    documents_uploaded: int
    message: str
